composer/datasets/in_context_learning_evaluation.py

```diff
@@ -85,582 +109,764 @@ def _make_padded_input(context_enc, continuation_enc, max_seq_len, pad_tok_id, p
     """
 
-    def _read_dataset(self, dataset: Dataset) -> List[Dict[str, str]]:
-        result = []
-        for example in dataset:
-            result.append({
-                'context': example['context'],
-                'answer': example['answer'],
-                'aliases': set([example['answer']] + example.get('aliases', [])),
-                'chain_of_thought': example.get('chain_of_thought', '')
-            })
-        return result
-
-    def __init__(
-        self,
-        dataset_uri: str,
-        tokenizer: Union[transformers.PreTrainedTokenizer, transformers.PreTrainedTokenizerFast],
-        max_seq_len: int,
-        pad_tok_id: int,
-        num_fewshot: int,
-        prompt_string: str,
-        example_delimiter: str,
-        continuation_delimiter: str,
-        destination_path: str,
-        question_prelimiter: str,
-        fewshot_random_seed: int,
-        cot_delimiter: str = '',
-    ):
-        try:
-            from datasets import load_dataset  # pyright: ignore [reportGeneralTypeIssues]
-        except ImportError as e:
-            raise MissingConditionalImportError(extra_deps_group='nlp',
-                                                conda_package='datasets',
-                                                conda_channel='conda-forge') from e
-        with dist.local_rank_zero_download_and_wait(destination_path):
-            if dist.get_local_rank() == 0:
-                get_file(dataset_uri, destination_path, overwrite=True)
-        dataset = load_dataset('json', data_files=destination_path, split='train', streaming=False)
-        self.samples = self._read_dataset(dataset)
-        self.samples = strip_data(self.samples)
+    def __init__(self,
+                 dataset_uri: str,
+                 tokenizer: transformers.PreTrainedTokenizerBase,
+                 max_seq_len: int,
+                 pad_tok_id: int,
+                 num_fewshot: int,
+                 fewshot_random_seed: int,
+                 prompt_string: str,
+                 example_delimiter: str,
+                 continuation_delimiter: str,
+                 destination_path: str,
+                 prelimiter: str = '',
+                 context_key: str = 'context',
+                 answer_key: str = 'answer',
+                 strip_dataset: bool = True,
+                 hf_loading_vars: Dict = None,
+                 hf_parsing_map: Dict = None,
+                 stacked_keys: List[str] = None,
+                 dont_split_keys: List[str] = None,
+                 list_split_keys: List[str] = None,
+                 normal_split_keys: List[str] = None,
+                 ):
+        
         self.tokenizer = tokenizer
+        self.prefix_space = _tokenizer_needs_prefix_space(self.tokenizer)
+
         self.max_seq_len = max_seq_len
         self.pad_tok_id = pad_tok_id
+        self.num_fewshot = num_fewshot
+        # TODO: check this is correct for all dataset types
```

Comment:
which vars are these todos referring to?

Reply:
`padding_side`. I have checked this now.